### ISSUE TITLE
Test on Firefox Android

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam2#52]

**Comment by dhowe** [comment]
status?

**Comment by CyrusSUEN** [comment]
Firefox for Android supports uBlock origin. It blocks ads on http://www.nytimes.com/ after installing the addon.

To run extension on Firefox:
1. get a signed XPI and the XPI file can be installed on Firefox for Android through AMO or opening and then install the signed XPI in Firefox 
2. through 'jpm-mobile' module and 'adb'
steps:
1. install Firefox nightly for Android from https://nightly.mozilla.org/
2. '$ npm install -g jpm-mobile'
3. '$ brew install android-platform-tools'
4. build and navigate to the Firefox build folder
5. '$ jpm init' and then fill in the info for package.json
6. find out the path of adb by '$ which adb'
7. connect the Android phone with adb by 'adb devices' and authenticate the operation on the phone
8. '$ jpm-mobile run --adb path/of/adb'

It should run on the Firefox nightly after step 8.
But you will encounter error because of bugs in jpm-mobile and need to apply patch
'$ npm install -g ncalexan/jpm-mobile#fixes'
https://github.com/mozilla-jetpack/jpm-mobile/issues/5#issuecomment-164010858
An another error from jpm-mobile
http://stackoverflow.com/a/32724075/1571343

It's just too much hassle to run it on Android because of the immaturity of jpm-mobile.
In theory valid Firefox addons should be able to run on Firefox for Android

> You can develop your add-on as normal, as long as you restrict yourself to the supported modules.

https://developer.mozilla.org/en-US/Add-ons/SDK/Tutorials/Mobile_development

-------------------------------
Chrome for Android doesn't support any kind of extensions.

> Chrome apps and extensions are currently not supported on Chrome for Android. We have no plans to announce at this time.

https://developer.chrome.com/multidevice/faq

-------------------------------
Opera and Opera Mini for Android also don't support extensions. (there was a dead project 3 years ago though)
http://forums.opera.com/discussion/1865635/add-extensions-to-opera-android/p1
http://forums.opera.com/discussion/1871155/extensions-in-opera-mini-android/p1

**Comment by dhowe** [comment]
We need to support Firefox for Android. Have you tested ADN on Firefox. If not, please do so.

**Comment by CyrusSUEN** [comment]
> Have you tested ADN on Firefox

I'm unable to test it because it requires a signed XPI for testing.
Or unless I can deal with the error http://stackoverflow.com/a/32724075/1571343 mentioned above which involves fixing jpm-mobile by hardcoding.

**Comment by dhowe** [comment]
Why can't you sign the xpi?

**Comment by CyrusSUEN** [comment]
> can't you sign the xpi?

I think that takes some days for AMO to review (and sign) it. Like we did in ADN1: https://github.com/dhowe/AdNauseam/issues/418

**Comment by dhowe** [comment]
on hold
